xbmc/video/dialogs/GUIDialogVideoVersion.h

```diff
@@ -0,0 +1,58 @@
+/*
+ *  Copyright (C) 2005-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include "MediaSource.h"
+#include "guilib/GUIDialog.h"
+#include "media/MediaType.h"
+#include "video/VideoDatabase.h"
+
+#include <memory>
+
+class CFileItem;
+class CFileItemList;
+class CVideoDatabase;
+
+class CGUIDialogVideoVersion : public CGUIDialog
+{
+public:
+  CGUIDialogVideoVersion(void);
+  ~CGUIDialogVideoVersion(void) override;
+  bool OnMessage(CGUIMessage& message) override;
+  void SetVideoItem(const CFileItem *item, bool playMode = false);
```

Comment:
Avoid raw pointers. change to const ref whenever possible. Here and elsewhere.

Avoid bool parameters. The calling code means nothing to the code reader (SetVideoVersion(item, true) means what?) Introduce two methods, each with a speaking name.

Avoid default parameters in public methods. Thise are proven to introduce hard to find errors, for example if the code gets refactored in the future.